File: BaseballTheater/Shared/Scripts/MlbDataServer/DataStructures/highlight.ts
namespace Theater
{
	export interface IHighlight
	{
		type: string;
		id: number;
		team_id: number;
		date: Date;
		headline: string;
		blurb: string;
		duration: string;
		url: IUrl[];
		thumb: string;
		thumbnails: IThumbnails;
		keywords: Keyword[];
		condensed: boolean;
		recap: boolean;
		"top-play": string;

		isPlaying: boolean;
	}

	export interface IThumbnails
	{
		thumb: IThumb[];
	}

	export interface IThumb
	{
		__text: string;

	}

	export interface IUrl
	{
		type: string;
		__text: string;
	}
}

File: MlbDataServer/Contracts/Highlight.cs
using System;
using System.Globalization;
using System.Linq;
using System.Xml.Serialization;

namespace MlbDataServer.Contracts
{
	[Serializable, XmlRoot("media")]
	public class Highlight
	{
		[XmlAttribute("type")]
		public string type { get; set; }

		[XmlAttribute("id")]
		public long id { get; set; }

		[XmlAttribute("date")]
		public string date { get; set; }

		[XmlElement("player")]
		public Players players { get; set; }

		[XmlElement("team")]
		public Team team { get; set; }

		public DateTime dateObj => DateTime.Parse(date, CultureInfo.InvariantCulture);

		[XmlElement("headline")]
		public string headline { get; set; }

		[XmlElement("blurb")]
		public string blurb { get; set; }

		[XmlElement("bigblurb")]
		public string bigblurb { get; set; }

		[XmlElement("duration")]
		public string duration { get; set; }

		[XmlElement("url")]
		public string[] url { get; set; }

		[XmlElement("thumb")]
		public string thumb { get; set; }

		[XmlArray("thumbnails")]
		[XmlArrayItem("thumb")]
		public string[] thumbnails { get; set; }

		[XmlArray("keywords")]
		[XmlArrayItem("keyword")]
		public Keyword[] keywords { get; set; }

		[XmlAttribute("condensed")]
		public bool condensed { get; set; }

		[XmlAttribute("recap")]
		public bool recap { get; set; }
	}

}